.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,226 @@
+on:
+  workflow_call:
+    inputs:
+      k8sVersion:
+        description: version of k3s to use or "kind" and "kindIpv6"
+        type: string
+        default: v1.28.1-k3s1
+        required: false
+      parallelism:
+        description: level of parallelization
+        type: string
+        default: '1'
+        required: false
+      target:
+        description: makefile target without e2e prefix
+        type: string
+        default: ""
+        required: false
+      arch:
+        description: The golang arch
+        type: string
+        default: amd64
+        required: false
+      cniNetworkPlugin:
+        description: The CNI networking plugin to use [flannel | calico]
+        type: string
+        default: flannel
+        required: false
+      legacyKDS:
+        description: if should run tests with new implementation of KDS
+        type: string
+        default: 'false'
```

Comment:
Remove the defaults here too and make required everything. It will be easier to debug when something is wrong (it's only possibly set in 1 place)